packages/docusaurus-theme-classic/src/theme/AnnouncementBar/index.tsx

```diff
@@ -13,9 +13,7 @@ import useUserPreferencesContext from '@theme/hooks/useUserPreferencesContext';
 import styles from './styles.module.css';
 
 function AnnouncementBar(): JSX.Element | null {
-  const {
-    siteConfig: {themeConfig: {announcementBar = {}} = {}} = {},
-  } = useDocusaurusContext();
+  const {announcementBar} = useDocusaurusContext().siteConfig.themeConfig;
   const {content, backgroundColor, textColor, isCloseable} = announcementBar;
```

Comment:
announcementBar may be null here :)

Reply:
what will be the right way to do this?

Reply:
I'd say just something like that:

```tsx
function AnnouncementBar(): JSX.Element | null {
  const {announcementBar} = useDocusaurusContext().siteConfig.themeConfig;
  if (!announcementBar) {
    return null;
  }
  const {content, backgroundColor, textColor, isCloseable} = announcementBar;
  
  // ...
}
```

When you destructure themeConfig, you have to look at the validation schema. If an attribute does not have any default value, that means it can be null, and you should not destruture if before checking the existence of such attribute

Reply:
Also something that could be useful is to create a hook like "useThemeConfig" with TS typing. If you don't know typing this can be done in another PR anyway

Reply:
Thanks. While doing it, eslint throws `error  React Hook "useUserPreferencesContext" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks`

Reply:
In all case you have to follow the hooks rules 😜

Reply:
cool :)